src/user/user.service.spec.ts

```diff
@@ -1,18 +1,248 @@
+      expect(result._id).toBe(userCollection[0]._id);
+    });
+
+    it('should throw error with unknown phone number', async () => {
+      jest.spyOn(userModel, 'findOne').mockReturnValue(undefined);
+      const result = await userService.findByPhoneNumber('-1');
+      expect(result).toBe(undefined);
+    });
   });
 
-  it('should be defined', () => {
-    expect(service).toBeDefined();
+  describe('find user either by phone number or email', () => {
+    it('should return user with corresponding inputs', async () => {
+      jest.spyOn(userModel, 'findOne').mockReturnValue(userCollection[0] as any);
+      const result1 = await userService.findUserByPhoneNumberOrEmail('1', undefined);
+      const result2 = await userService.findUserByPhoneNumberOrEmail(undefined, 'test1@email.com');
+      expect(result1._id).toBe(result2._id);
+    });
+
+    it('should not return user with corresponding inputs', async () => {
+      jest.spyOn(userModel, 'findOne').mockReturnValue(undefined as any);
+      const result1 = await userService.findUserByPhoneNumberOrEmail('-1', undefined);
+      const result2 = await userService.findUserByPhoneNumberOrEmail(undefined, 'test999@email.com');
+      expect(result1 && result2).toBeFalsy();
+    });
+  });
+
+  describe('register customer user', () => {
+    const user: User = {
+      _id: new mongoose.Types.ObjectId(),
+      phoneNumber: '5',
+      email: 'test5@email.com',
+      username: '5',
+      password: 'password',
+      verificationStatus: VerificationStatus.NOT_VERIFIED,
+      accountType: 'CUSTOMER',
+      profileImage: 'string',
+      firstName: 'Mr.5', // type OnePieceReference
+      lastName: 'Gem',
+    };
+
+    it('should not create new user', async () => {
+      try {
+        await userService.registerCustomerUser(userCollection[0] as any);
+        jest.spyOn(userModel, 'findOne').mockReturnValue(userCollection[0] as any);
+      } catch (e) {
+        expect(e).toBeInstanceOf(HttpException);
+      }
+    });
+    it('should create new user', async () => {
+      jest.spyOn(userModel, 'findOne').mockReturnValue(undefined);
+      const result = await userService.registerCustomerUser(user as any);
+      // verify result.jwt
+    });
+  });
+
+  // TODO: Find a good solution for userModel.save() is undefined...
+  describe('should set phone verification status', () => {
+    it('should set verification status to true', async () => {
+      jest.spyOn(userModel, 'findById').mockReturnValue({
+        select: jest.fn().mockReturnValue(userCollection[0]),
+      } as any);
+
+      // jest.spyOn(userModel, 'create').mockImplementationOnce(() =>
+      //   Promise.resolve({
+      //     status: 200,
+      //   }),
+      // );
+      // const result = await userService.setUserPhoneVerificationStatus(
+      //   userCollection[0]._id.toString(),
+      //   VerificationStatus.VERIFIED,
+      // );
+
+      // expect(result.status).toBe(200);
+    });
```

Comment:
Still using?

Reply:
Yes, leave it here until I can find a solution to userModel.create().
I have noted in TODO comment above it.